shared/src/main/kotlin/watchfaces/dslclock/DSLClockComplication.kt

```diff
@@ -0,0 +1,355 @@
+            )
+        }
+    }
+
+    OClockCanvas {
+        rotate(degrees = 0f) {
+            drawText(
+                brush = style.brush ?: SolidColor(style.color),
+                textLayoutResult = textLayoutResult,
+                topLeft = Offset(
+                    x = 120f,
+                    y = (center.y * 0.8f) - (textLayoutResult.size.height / 2f)
+                )
+            )
+        }
+    }
+
+}
+
+
+@Composable
+private fun HeartRate(style: TextStyle) {
+
+    val context = LocalContext.current
+    val isAmbient by LocalIsAmbient.current
+    var hr by remember { mutableIntStateOf(0) }
+
+    val sensorManager = context.getSystemService(Context.SENSOR_SERVICE) as SensorManager
+    val hrCounter = sensorManager.getDefaultSensor(Sensor.TYPE_HEART_RATE)
+
+    val heartRateListener = object : SensorEventListener {
+        override fun onSensorChanged(event: SensorEvent?) {
+            if (isAmbient.not()) {
+                try {
+                    hr = event?.values?.get(0)?.toInt() ?: 0
+                } catch (_: IllegalStateException) {
+                }
+
+            }
+        }
+
+        override fun onAccuracyChanged(sensor: Sensor?, accuracy: Int) {
+
+        }
+    }
+
+    hrCounter.let { sensor ->
+        when (isAmbient) {
+            true -> sensorManager.unregisterListener(heartRateListener)
+            false -> {
+                sensorManager.registerListener(
+                    heartRateListener,
+                    sensor,
+                    SensorManager.SENSOR_DELAY_NORMAL
+                )
+            }
+        }
+
+    }
+
+
+    @OptIn(InternalComposeOClockApi::class)
+    val measurer = LocalTextMeasurerWithoutCache.current
+
+    val textLayoutResult by remember {
+        derivedStateOf {
+            measurer.measure(
+                "heart rate = $hr bps",
+                style
+            )
+        }
+    }
+
+    OClockCanvas {
+        rotate(degrees = 0f) {
+            drawText(
+                brush = style.brush ?: SolidColor(style.color),
+                textLayoutResult = textLayoutResult,
+                topLeft = Offset(
+                    x = 120f,
+                    y = center.y - (textLayoutResult.size.height / 2f)
+                )
+            )
+        }
+    }
+
+}
+
+
+@Composable
+private fun Steps(style: TextStyle) {
+
+    var steps by remember { mutableIntStateOf(-1) }
+
+    @OptIn(InternalComposeOClockApi::class)
+    val measurer = LocalTextMeasurerWithoutCache.current
+
+    val textLayoutResult by remember {
+        derivedStateOf {
+            measurer.measure(
+                "steps = $steps",
+                style
+            )
+        }
+    }
+
+    OClockCanvas {
+        rotate(degrees = 0f) {
+            drawText(
+                brush = style.brush ?: SolidColor(style.color),
+                textLayoutResult = textLayoutResult,
+                topLeft = Offset(
+                    x = 120f,
```

Comment:
Hardcoded pixel size that is not dependent on the canvas size is gonna be problematic since not all screens have the same density (and resolution).